src/CreeDictionary/API/search/test_eip_crk.py

```diff
@@ -1,24 +1,72 @@
+import pytest
+
 from CreeDictionary.API.models import Wordform
 from CreeDictionary.API.search.core import SearchRun
 from CreeDictionary.API.search.eip import EipSearch
 from CreeDictionary.API.search.types import Result
 
 
-def test_eip_search(db):
-    search_run = SearchRun("they crawled")
+@pytest.mark.parametrize(
+    ("search", "params"),
+    [
+        [
+            "they crawled",
+            {
+                "expected_query_terms": ["crawls"],
+                "expected_new_tags": ["+V", "+AI", "PV/ki+", "+Ind", "+3Pl"],
+                "lemma": "pimitâcimow",
+                "expected_inflection": "kî-pimitâcimowak",
+            },
+        ],
+        [
+            "they saw me",
+            {
+                "expected_query_terms": ["see"],
+                "expected_new_tags": ["+V", "+TA", "PV/ki+", "+Ind", "+3Pl", "+1SgO"],
+                "lemma": "wâpamêw",
+                "expected_inflection": "nikî-wâpamikwak",
+            },
+        ],
+        [
+            "my little bears",
+            {
+                "expected_query_terms": ["bear"],
+                "expected_new_tags": ["+N", "+Der/Dim", "+Px1Sg", "+Pl"],
+                "lemma": "maskwa",
+                "expected_inflection": "nimaskomisak",
+            },
+        ],
+        [
+            "they see",
+            {
+                "expected_query_terms": ["see"],
+                "expected_new_tags": ["+V", "+AI", "+Ind", "+3Pl"],
+                "lemma": "wâpiw",
+                "expected_inflection": "wâpiwak",
+            },
+        ],
+    ],
+)
+def test_eip_search(db, search, params):
+    search_run = SearchRun(search)
     eip_search = EipSearch(search_run)
     eip_search.analyze_query()
-    assert search_run.query.query_string == "crawls"
-    assert search_run.query.query_terms == ["crawls"]
-    assert eip_search.new_tags == ["+V", "+AI", "PV/ki+", "+Ind", "+3Pl"]
+    assert search_run.query.query_terms == params["expected_query_terms"]
+    assert search_run.query.query_string == " ".join(params["expected_query_terms"])
+    assert eip_search.new_tags == params["expected_new_tags"]
 
-    lemma1 = Wordform.objects.get(text="pimitâcimow", is_lemma=True)
+    lemma1 = Wordform.objects.get(text=params["lemma"], is_lemma=True)
 
     search_run.add_result(
-        Result(wordform=lemma1, target_language_keyword_match=["crawls"])
+        Result(
+            wordform=lemma1,
+            target_language_keyword_match=params["expected_query_terms"],
+        )
     )
 
     eip_search.inflect_search_results()
 
-    # kî-pimitâcimowak = “they crawled”
-    assert list(search_run.unsorted_results())[0].wordform.text == "kî-pimitâcimowak"
+    assert (
+        list(search_run.unsorted_results())[0].wordform.text
```

Comment:
It will always be the first, unsorted result?

Reply:
Yeah, that’s what we’re testing. We create a search run, add a single result to it, tell the EIP search to do its inflection magic, and make sure that the single result has then been updated.